Marlin/src/inc/Conditionals_LCD.h

```diff
@@ -860,16 +860,30 @@
 #endif
 #if EITHER(HAS_ABL_OR_UBL, MESH_BED_LEVELING)
   #define HAS_LEVELING 1
+
+  #define DEACTIVATE_LVL  0
+  #define ACTIVATE_LVL    1
+  #define RESTORE_LVL     2
+
   #if DISABLED(AUTO_BED_LEVELING_UBL)
     #define PLANNER_LEVELING 1
   #endif
+  #if LEVELING_AFTER_G28 != DEACTIVATE_LVL
+    #define CHANGES_LEVELING_AFTER_G28 1
```

Comment:
If `LEVELING_AFTER_G28` is set as `RESTORE_LVL`, then will it be set as `ACTIVATE_LVL`? (in G34 and G35 too)

Reply:
No you have to program LEVELING_AFTER_G34 and LEVELING_AFTER_G35 for desired action

Reply:
Is it, if `LEVELING_AFTER_G28` isn't equal to 0, then set `CHANGES_LEVELING_AFTER_G28` to 1? So either `LEVELING_AFTER_G28` is 1, or 2, `CHANGES_LEVELING_AFTER_G28` will be 1?
Am I reading it wrong?

Reply:
Note that RESTORE_LVL is not the same as ACTIVATE_LVL. Former will restore leveling you had before G28, G34 or G35. latter will force leveling on

Reply:
That is why the code looks weird to me.

Reply:
CHANGES_LEVELING_AFTER_Gxx is used to change leveling after command. State is forced to 1 or to previous depending on LEVELING_AFTER_G?? value:
`TERN_(CHANGES_LEVELING_AFTER_G28, const bool leveling_restore_state = LEVELING_AFTER_G28 == ACTIVATE_LVL || planner.leveling_active);`

Reply:
Ok, I see it now, it is a bit twisted but works.
Would `TERN_` work if `CHANGES_LEVELING_AFTER_G28` is equal to 2? If yes, then maybe this would be better:
```
#if LEVELING_AFTER_G28 != DEACTIVATE_LVL
    #define CHANGES_LEVELING_AFTER_G28 LEVELING_AFTER_G28
```

Reply:
CHANGES_LEVELING_AFTER_G28 is never 2 its only "defined as 1" or undefined

Reply:
Not sure, TERN_ is so nested that I lose path to the end

Reply:
I suspect it only works with 1, 0, true, false

Reply:
If `TERN_` works the same way as other bool checks, then `0 = false` and `anything that isn't 0 = true`, but this needs confirmation from someone else.

Reply:
Now sure it works like a boolean test directly. I think it works by replacing text and comparing such derived text
  #define TERN_(O,A)          _TERN(_ENA_1(O),,A)     // OPTION converted to A or '<nul>'
  #define _ENA_1(O)           _ISENA(CAT(_IS,CAT(ENA_, O)))
  #define _ISENA(V...)        IS_PROBE(V)
  #define IS_PROBE(V...) SECOND(V, 0)     // Get the second item passed, or 0

Reply:
[ternary-macros](https://github.com/MarlinFirmware/MarlinDocumentation/blob/master/_development/coding_standards.md#)

Reply:
yep but they emit something if enabled something else if not, but what is "enabled" boolean or what? Not defined in above link so only thing should be ask or test...

Reply:
That site/file might be a bit incomplete.

Reply:
prefer IF_ENABLED() /  IF_DISABLED(cond, <code>) for "if" without else, keep TERN for If/Else (3 params)

Reply:
it's not a "really" enabled or disabled option but ok

Reply:
@GMagician I think this can be set as resolved, because at the moment TERN isn't ready to have 2 as a true value.